client/core/wallet.go

```diff
@@ -0,0 +1,107 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package core
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"decred.org/dcrdex/client/asset"
+	"decred.org/dcrdex/dex"
+)
+
+// xcWallet is a wallet.
+type xcWallet struct {
+	asset.Wallet
+	connector *dex.ConnectionMaster
+	AssetID   uint32
+	mtx       sync.RWMutex
+	lockTime  time.Time
+	hookedUp  bool
+	balance   uint64
+	balUpdate time.Time
+	encPW     []byte
+	address   string
+}
+
+// Unlock unlocks the wallet.
+func (w *xcWallet) Unlock(pw string, dur time.Duration) error {
+	err := w.Wallet.Unlock(pw, dur)
+	if err != nil {
+		return err
+	}
+	w.mtx.Lock()
+	w.lockTime = time.Now().Add(dur)
+	w.mtx.Unlock()
+	return nil
+}
+
+// lock the wallet, setting the lockTime to the time when the wallet will be
+// locked.
```

Comment:
I know it hasn't changed, but the comment is a bit confusing since it's setting `lockTime` to the zero `time.Time` value for the purposes of having `time.(Time).After(now)` always return true in `unlocked` and `state`.  Seems like the current comment is describing what `Unlock` does.

`Lock` cap.